### PR TITLE
fix: Add aromatic aluminium token

### LINKED ISSUE #3697: Lower case symbols in SMILES for bracket atoms in aromatic rings
Tested with rdkit versions (2020.09 and 2020.03), python 3.7

Minimal example:

```python
from rdkit import Chem
smi = '[Al+]1cccccccccc1'
mol = Chem.MolFromSmiles(smi)
print(Chem.MolToSmiles(mol))
# prints 'c1ccccc[al+]cccc1'
```

`[Al+]` turned into `[al+]` when converted to a SMILES and is not readable anymore. 

It seems that atoms in an aromatic ring get a lower case symbol independent of their type. 

Example PubChem entry with this problem ([https://pubchem.ncbi.nlm.nih.gov/compound/71310059](https://pubchem.ncbi.nlm.nih.gov/compound/71310059))

```python
smi = 'C1=CC=C(C=C1)OC2=CC3=C4NC(=C3C=C2)NC5=C6C=C(C=CC6=C7N5[Al+]N8C(=C9C=CC(=CC9=C8NC1=C2C=CC(=CC2=C(N1)N7)OC1=CC=CC=C1)OC1=CC=CC=C1)N4)OC1=CC=CC=C1.[Cl-]'
mol = Chem.MolFromSmiles(smi)
print(Chem.MolToSmiles(mol))
#prints '[Cl-].c1ccc(Oc2ccc3c4[nH]c([nH]c5c6ccc(Oc7ccccc7)cc6c6[nH]c7[nH]c([nH]c8c9ccc(Oc%10ccccc%10)cc9c([nH]4)n8[al+]n56)c4cc(Oc5ccccc5)ccc74)c3c2)cc1'
```

### PR DESCRIPTION
**Pull request by iwyoo**
#### Reference Issue
Fixes #3697

#### What does this implement/fix? Explain your changes.
By Huckel's rule, rdkit sometimes converts `[Al+]` to `[al+]`, but current rdkit cannot read `[al` because there is no lex rule for it.

### CODE REVIEW

**Comment by greglandrum** [comment]
@iwyoo , thanks for the contribution.
I think a better fix for this would be to avoid generating the aromatic token for Al in the first place.
So the modification would be to SmilesWrite.cpp, around line 231, to make sure that it only changes the first letter in the atomic symbol to be aromatic if the atomic number is 5, 6, 7, 8, 14, 15, 16, 33, 34, or 52

**Comment by iwyoo** [comment]
@greglandrum Thanks! I just deleted the previous changes and applied the correct revision to SmilesWrite.cpp you mentioned.

**Review by greglandrum — CHANGES_REQUESTED** [review]
Thanks for updating the PR.

The one piece which is missing for this to be added is a test to demonstrate that the changes work (and make sure that if the behavior changes in the future we will detect it).

Please add the new test(s) to `v2catch_tests.cpp` in the `SmilesParse` directory; you can use the tests that are there or in `catch_tests.cpp` as a template for what the tests should look like.

Let me know if you have any questions.

**Comment by iwyoo** [comment]
@greglandrum Thank you for your kind comment! I wrote a test that solves the issue(#3697). I'm not sure whether the test is enough or not. If you need anything more, let me know!

File: Code/GraphMol/SmilesParse/v2catch_tests.cpp [hunk 1]
```diff
@@ -66,4 +66,10 @@ TEST_CASE("v2 basics") {
     auto bnd = SmilesParse::BondFromSmarts("@");
     REQUIRE(bnd);
   }
-}
\ No newline at end of file
+  {
+    auto mol = SmilesParse::MolFromSmiles("[Al+]1cccccccccc1");
```
Comment (greglandrum) [inline Code/GraphMol/SmilesParse/v2catch_tests.cpp:70]: This should be its own `TEST_CASE`, something like this:

```suggestion
}
TEST_CASE("handling of aromatic Al in SMILES"){
  SECTION("basics") {
      auto mol = SmilesParse::MolFromSmiles("[Al+]1cccccccccc1");
```
(note, I haven't tested that and may have missed some punctuation)

**Review by greglandrum — CHANGES_REQUESTED** [review]
The test you added looks good, but it should be its own `TEST_CASE` that references the name of the bug you are fixing.

File: Code/GraphMol/SmilesParse/v2catch_tests.cpp (hunk 1 above)
Comment (iwyoo) [reply Code/GraphMol/SmilesParse/v2catch_tests.cpp]: Thanks for the suggestion! I applied.

**Review by greglandrum — APPROVED** [review]
LGTM

**Comment by greglandrum** [comment]
Thanks for the contribution @iwyoo !

**Pull request merged**